src/nvim/os/pty_process_win.c

```diff
@@ -192,8 +229,18 @@ void pty_process_resize(PtyProcess *ptyproc, uint16_t width,
                         uint16_t height)
   FUNC_ATTR_NONNULL_ALL
 {
-  if (ptyproc->winpty_object != NULL) {
-    winpty_set_size(ptyproc->winpty_object, width, height, NULL);
+  if (ptyproc->type == PTY_TYPE_CONPTY
+      && ptyproc->pty_object.conpty_object != NULL) {
+    assert(width <= SHRT_MAX);
+    assert(height <= SHRT_MAX);
+    COORD size = { (int16_t)width, (int16_t)height };
+    if (pResizePseudoConsole(
+        ptyproc->pty_object.conpty_object->pty, size) != S_OK) {
+      ELOG("ResizePseudoConsoel failed: error code: %d",
+           os_translate_sys_error((int)GetLastError()));
+    }
+  } else if (ptyproc->pty_object.winpty_object != NULL) {
+    winpty_set_size(ptyproc->pty_object.winpty_object, width, height, NULL);
```

Comment:
can we hide the win32 API code in something similar to `winpty_set_size` (preferably in `os_win_conpty.c` or similar isolated module)? We want to minimize the "scattered" changes.